File: src/net/theharrisoncrafter/Commands/Throw.java
package net.theharrisoncrafter.Commands;

import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class Throw implements CommandExecutor{
	public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args){		
		if(sender instanceof Player){
			if(sender.hasPermission("GodCraft.Throw")){
				Player targetPlayer;
				String item;
				
				if(args.length < 1 || args.length > 2){
					return false;
				}
				
				if(args.length > 1){
					targetPlayer = sender.getServer().getPlayer(args[0]);
					item = args[1];
				}
				else{
					targetPlayer = (Player) sender;
					item = args[0];
				}
				
				if(targetPlayer == null){
					sender.sendMessage(ChatColor.RED + "That player is not online!");
					return true;
				}
				
				if(item.toLowerCase().equals("egg")){
					targetPlayer.throwEgg();
				}
				else if(item.toLowerCase().equals("snowball")){
					targetPlayer.throwSnowball();
				}
				else if(item.toLowerCase().equals("arrow")){
					targetPlayer.shootArrow();
				}
				
				return true;
			}
			else{
				sender.sendMessage(ChatColor.RED + "You do not have permissions to do that!");
				return true;
			}
		}
		else{
			Player targetPlayer;
			
			if(args.length < 2 || args.length > 2){
				return false;
			}
			
			targetPlayer = Bukkit.getPlayer(args[0]);
			
			if(targetPlayer == null){
				sender.sendMessage(ChatColor.RED + "That player is not online!");
				return true;
			}
			
			if(args[1].toLowerCase().equals("egg")){
				targetPlayer.throwEgg();
			}
			else if(args[1].toLowerCase().equals("snowball")){
				targetPlayer.throwSnowball();
			}
			else if(args[1].toLowerCase().equals("arrow")){
				targetPlayer.shootArrow();
			}
			
			return true;
		}
	}
}

File: src/main/kotlin/co/orre/godcraft/Commands/Throw.kt
package co.orre.godcraft.Commands

import co.orre.godcraft.God
import org.bukkit.Bukkit
import org.bukkit.ChatColor as CC
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Arrow
import org.bukkit.entity.Egg
import org.bukkit.entity.Player
import org.bukkit.entity.Snowball

class Throw(val plugin: God) : CommandExecutor {
    override fun onCommand(sender: CommandSender, cmd: Command, commandLabel: String, args: Array<String>): Boolean {
        if (sender is Player) {
            if (sender.hasPermission("godcraft.throw")) {
                val targetPlayer: Player?
                val item: String

                if (args.isEmpty() || args.size > 2) return false

                if (args.size == 1) {
                    targetPlayer = sender
                    item = args[0].toLowerCase()
                } else {
                    targetPlayer = sender.getServer().getPlayer(args[0])
                    item = args[1].toLowerCase()
                }

                if (targetPlayer == null) {
                    sender.sendMessage("${CC.RED}That player could not be found!")
                    return true
                }

                when (item) {
                    "egg" -> targetPlayer.launchProjectile(Egg::class.java)
                    "snowball" -> targetPlayer.launchProjectile(Snowball::class.java)
                    "arrow" -> targetPlayer.launchProjectile(Arrow::class.java)
                    else -> return false
                }
                plugin.logDebug("${sender.name} made ${targetPlayer.name} shoot a projectile (type: $item)")
                return true
            }
            sender.sendMessage("${CC.RED}You do not have permissions to do that!")
            return true
        }
        if (args.size != 2) return false
        val targetPlayer: Player? = Bukkit.getPlayer(args[0])
        if (targetPlayer == null) {
            sender.sendMessage("${CC.RED}That player could not be found!")
            return true
        }

        val item = args[1].toLowerCase()
        when (item) {
            "egg" -> targetPlayer.launchProjectile(Egg::class.java)
            "snowball" -> targetPlayer.launchProjectile(Snowball::class.java)
            "arrow" -> targetPlayer.launchProjectile(Arrow::class.java)
            else -> return false
        }
        plugin.logDebug("${sender.name} made ${targetPlayer.name} shoot a projectile (type: $item)")
        return true
    }
}
